Src/World/Timing.cpp

```diff
@@ -1,70 +1,78 @@
 #include "Timing.h"
 
-#include <Math/Math.h>
-
 
 // The maximum amount of time the accumulator can store.
-constexpr double MAX_ACCUMULATED_SECONDS = 3.0;
+static constexpr double MAX_ACCUMULATED_SECONDS = 3.0;
+using Clock = std::chrono::high_resolution_clock;
+
 
-Timing::Timing(int tickrate) {
-    timeStep = 1.0 / tickrate;
-    accumulatedSeconds = 0.0;
-    secondToLastFixedTime = 1.0;
-    lastFixedTime = 1.0;
+Timing::Timing(int tickrate, int framerate) {
```

Comment:
What does setting the framerate even mean here? The framerate would ideally be clamped by vsync, or left uncapped.

Reply:
Uncapped felt .. problematic.
Main benefit I can think of is to smooth out lag frames from angelscript code, but I don't know what vsync clamping is or does.
Up to you, it is easily changed or modified. Or maybe give angelscript rendering frames, but leave the other stuff to vsync clamping? There's a lot of ways to go about this.

Reply:
> Uncapped felt .. problematic.

This is ridiculous. If this code doesn't actually do anything for any good reason, do not keep it.

Reply:
Was your accumulator intended to fix for lag frames?
It appeared to have the potential problem of, during lagframes or slowdowns, the accumulator will accumulate up to 3 seconds worth of ticks, and when given the next available opportunity, it will immediately execute all of those ticks at once.

I was also concerned that it could've been much more accurate because it appeared to be using milliseconds instead of nanoseconds, which I found had significant margins for error.
I believe this approach is ultimately superior because it deals directly in high-precision time points and always executes at the fastest possible rates in a smooth manner, instead of by using time differences in a very strange accumulator.

I would be grateful to understand more of how your timing system works and how it is intended to work.